sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/common/GcpIoPipelineOptionsRegistrar.java

```diff
@@ -38,6 +39,7 @@ public Iterable<Class<? extends PipelineOptions>> getPipelineOptions() {
         .add(PubsubOptions.class)
         .add(FirestoreOptions.class)
         .add(TestBigQueryOptions.class)
+        .add(BigtableTestOptions.class)
```

Comment:
If this is the only place that refers to BigtableTestOption class, it's not necessary to move the Option class. We can register the option in the test class.

Reply:
Maybe there's an alternate solution. I want to be able to pass in "instanceId" as a parameter to the IT. But gcp project defines `integrationTest` to explicitly take a few parameters and pass them to beam `beamTestPipelineOptions`. So I added `instanceId`. But, unless `instanceId` is always defined (by registering it for all ITs), other tests fail because they don't have `instanceId` defined but we're passing `instanceId`.

So this ended up being the solution. Happy to try something else.

Reply:
In each IT class, one can get a TestPipelineOptions instance like this

https://github.com/apache/beam/blob/8a28100c47b15cf04ec76aba2e1d00fd760aeebd/sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOJsonIT.java#L69

and then one can use testOptions.as(BigtableTestOptions.class) to get BigtableTestOptions.

Current Bigtable ITs is doing similar things and it should suffice.

https://github.com/apache/beam/blob/ba0259de7f9f2d9e06108de20ef57b16ef8e3cc4/sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/changestreams/it/BigtableChangeStreamIT.java#L80C5-L80C12

> unless instanceId is always defined (by registering it for all ITs), other tests fail because they don't have instanceId defined but we're passing instanceId.

That is because "registering it for all ITs". If we do not register it for all ITs then irrelevant tests won't fail.